### PR TITLE
Refactor views

### PR DESCRIPTION
**Pull request by jimrandomh**
Refactor the views system a bit. Makes various places around the code-base that get the default view, use the view system via a more purposeful entry point instead. This disarms a footgun which I ran into in another branch, which was the motivation for doing this now: the thing you got back from `Collection.getParameters({}).selector` was *almost* a ready-to-use mongodb selector, except it can have `viewFieldNullOrMissing` in it if the particular collection's default view contains that. Existing usages lucked out of hitting this case, but some new code did.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1203618950006734) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/lib/utils/viewUtils.ts [hunk 1]
```diff
@@ -13,3 +22,114 @@ export function describeTerms(terms: ViewTermsBase) {
...
+  logger('getParameters(), final parameters:', parameters);
+  return parameters;
+}
+
+function replaceSpecialFieldSelectors(selector: any): any {
+  let result: any = {};
+  for (let key of Object.keys(selector)) {
+    if (_.isEqual(selector[key], viewFieldNullOrMissing)) {
+      // Put an explicit null in the selector. In mongodb-query terms, this means
+      // no constraint on the given field.
```
Comment (jpaddison3) [inline packages/lesswrong/lib/utils/viewUtils.ts:121]: > no constraint

This isn't correct to my reading. I read that comment as saying "'anything goes' for this field." However, the null is explicitly constraining the results to be either null or missing.

File: packages/lesswrong/lib/utils/viewUtils.ts [hunk 2]
```diff
@@ -13,3 +22,114 @@ export function describeTerms(terms: ViewTermsBase) {
...
+  // limit number of items to 1000 by default
+  const maxDocuments = maxDocumentsPerRequestSetting.get();
+  const limit = terms.limit || parameters.options.limit;
+  parameters.options.limit = !limit || limit < 1 || limit > maxDocuments ? maxDocuments : limit;
+
+  logger('getParameters(), final parameters:', parameters);
+  return parameters;
+}
+
+function replaceSpecialFieldSelectors(selector: any): any {
```
Comment (jpaddison3) [inline packages/lesswrong/lib/utils/viewUtils.ts:116]: Docstring please.

File: packages/lesswrong/lib/utils/viewUtils.ts [hunk 3]
```diff
@@ -13,3 +22,114 @@ export function describeTerms(terms: ViewTermsBase) {
   else
     return viewName;
 }
+
+export function viewTermsToQuery<N extends CollectionNameString>(collectionName: N, terms: ViewTermsByCollectionName[N], apolloClient?: any, resolverContext?: ResolverContext) {
```
Comment (jpaddison3) [inline packages/lesswrong/lib/utils/viewUtils.ts:26]: Also docstring please

**Review by jpaddison3 — APPROVED** [review]
Great work. I love this, the extraction of `replaceSpecialFieldSelectors` and use in `getDefaultViewSelector` is Correct™️ in a very pleasing way.

ASTLAC. Please address comments about comments before merging.

File: packages/lesswrong/lib/utils/viewUtils.ts (hunk 1 above)
Comment (jimrandomh) [reply packages/lesswrong/lib/utils/viewUtils.ts:121]: Ah, yep, brain fart. (The `viewFieldNullOrMissing` identifier should've made it clear.)

**Pull request merged**